# Project 4 -- Jonathan Contreras

**TA Help:**
    
**Collaboration:** 

## Question 1

In [1]:
%%bash
# extracting all flight data that have the origin or destination at the Indianapolis airport into a indyflights.csv file
grep "IND" /anvil/projects/tdm/data/flights/subset/[12]*.csv >$HOME/indyflights.csv

In [2]:
%%bash
# extracting data contained in columns 17 and 18, sorting the data, counting unique data, sorting it numerically, and getting the most
# popular data.
cat $HOME/indyflights.csv | cut -d, -f17,18 | sort | uniq -c | sort -n | tail -10

  36350 MSP,IND
  36523 IND,DFW
  48922 ATL,IND
  48975 IND,ATL
  53331 STL,IND
  53789 DTW,IND
  54186 IND,STL
  55974 IND,DTW
  76554 ORD,IND
  77720 IND,ORD


Markdown notes and sentences and analysis written here.

## Question 2

In [3]:
%%bash
# extracting only the data with EAST stores and saving it withing a grocerydata.csv file
grep "EAST" /anvil/projects/tdm/data/8451/The_Complete_Journey_2_Master/5000_transactions.csv >$HOME/grocerydata.csv 

In [4]:
%%bash
# trying to figure out which column the date is in
cat $HOME/grocerydata.csv | tail

103041                  ,4525            ,26-DEC-17,3793066                     ,      2.49,         1,EAST   ,       104,2017
103065                  ,3768            ,27-DEC-17,2970080                     ,      2.99,         1,EAST   ,       104,2017
103072                  ,0193            ,27-DEC-17,0639663                     ,      2.99,         1,EAST   ,       104,2017
103072                  ,1539            ,27-DEC-17,3737283                     ,         1,         1,EAST   ,       104,2017
103137                  ,1507            ,29-DEC-17,0089283                     ,      4.69,         1,EAST   ,       104,2017
103159                  ,2332            ,29-DEC-17,0076870                     ,      2.14,         2,EAST   ,       104,2017
103334                  ,3734            ,28-DEC-17,4750235                     ,      6.87,         3,EAST   ,       104,2017
103372                  ,0006            ,29-DEC-17,6507918                     ,      3.99,         1,EAST   ,

In [5]:
%%bash
# extracting the date column, sorting it, counting unique data, sorting it numerically, getting the 10 most popular days
cat $HOME/grocerydata.csv | cut -d, -f3 | sort | uniq -c | sort -n | tail -10

   7362 26-MAR-17
   7414 05-FEB-17
   7420 08-JAN-17
   7451 17-APR-16
   7496 03-APR-16
   7650 15-APR-17
   7804 22-DEC-17
   7953 20-MAR-16
   8730 23-DEC-16
   8889 23-DEC-17


Markdown notes and sentences and analysis written here.

## Question 3

In [6]:
%%bash
# extracting the head of the data in the 8th column
cat /anvil/projects/tdm/data/election/itcont*.txt | cut -d'|' -f8 | head

MCKENNON, K R
OREFFICE, P
DOWNEY, J
BLAIR, E
BLANCHARD, JOHN A
CRAMER, JOHN H
MCHUGH, KEVIN
NOHA, EDWARD J
RYCROFT, DONALD C
VANDERSLICE, WILLIAM D


In [7]:
%%bash
# extracting the head of the data now in the 1st and 8th column
cat /anvil/projects/tdm/data/election/itcont*.txt | cut -d'|' -f8 | cut -d, -f1 | head

MCKENNON
OREFFICE
DOWNEY
BLAIR
BLANCHARD
CRAMER
MCHUGH
NOHA
RYCROFT
VANDERSLICE


In [8]:
%%bash
# extracting the tail of the data in the 1st and 8th column, sorting it, counting unique data, sorting numerically
cat /anvil/projects/tdm/data/election/itcont*.txt | cut -d'|' -f8 | cut -d, -f1 | sort | uniq -c | sort -n | tail -10

 496200 TAYLOR
 603308 WILSON
 695695 ANDERSON
 720429 DAVIS
 760715 WILLIAMS
 803227 JONES
 893166 BROWN
 899916 MILLER
1225266 JOHNSON
1654933 SMITH


Markdown notes and sentences and analysis written here.

## Question 4

In [9]:
%%bash
# extracting the head of the breweries data
cat /anvil/projects/tdm/data/beer/breweries.csv | head

id,name,city,state,country,notes,types
19730,Brouwerij Danny,Erpe-Mere,,BE,No notes at this time.,Brewery
32541,Coachella Valley Brewing Co,Thousand Palms,CA,US,No notes at this time.,"Brewery, Bar, Beer-to-go"
44736,Beef 'O' Brady's,Plant City,FL,US,No notes at this time.,"Bar, Eatery"
23372,Broadway Wine Merchant,Oklahoma City,OK,US,No notes at this time.,Store
35328,Brighton Beer Dispensary (DUPLICATE),Brighton,GB2,GB,Duplicate of https://www.beeradvocate.com/beer/profile/36929,"Bar, Eatery"
31561,Teddy's Tavern,Seattle,WA,US,No notes at this time.,"Bar, Beer-to-go"
35975,Modus Operandi Brewing Co.,Mona Vale,,AU,No notes at this time.,"Brewery, Bar, Eatery, Beer-to-go"
5618,Hops! Beer Restaurant & Pizza,Riccione (RN),,IT,No notes at this time.,"Brewery, Bar, Eatery"
30916,Kelly's Cellars,Belfast,GB5,GB,No notes at this time.,Bar


In [10]:
%%bash
# extracting the tail of the 3rd and 4th columns of breweries data
# then sorting the data, counting unique data, and sorting numerically
cat /anvil/projects/tdm/data/beer/breweries.csv | cut -d, -f3,4 | sort | uniq -c | sort -n | tail -10

    227 Minneapolis,MN
    238 Denver,CO
    274 Seattle,WA
    275 Brooklyn,NY
    290 Portland,OR
    302 London,GB2
    306 San Diego,CA
    427 New York,NY
    499 Philadelphia,PA
    512 Chicago,IL


Markdown notes and sentences and analysis written here.

## Question 5

In [11]:
%%bash
# extracting the amount of times 'salty' appears in the data. this data is ignoring case-sensitivity
# -i: ignores case-sensitive data
# -c: counts the data
grep -i -c "salty" /anvil/projects/tdm/data/icecream/combined/reviews.csv 

350


In [12]:
%%bash
# extracting the amount of times 'sweet' appears in the data. this data is ignoring case-sensitivity
# -i: ignores case-sensitive data
# -c: counts the data
grep -i -c "sweet" /anvil/projects/tdm/data/icecream/combined/reviews.csv

2080


In [13]:
%%bash
# extracting the amount of times 'chocolate' appears in the data. this data is making sure only 'chocolate' appears, so it is case-sensitive
# -c: counts the data
grep -c "chocolate" /anvil/projects/tdm/data/icecream/combined/reviews.csv

4754


In [14]:
%%bash
# extracting the amount of times 'chocolate' appears in the data. this data is ignoring case-sensitivity
# -i: ignores case-sensitive data
# -c: counts the data
grep -i -c "chocolate" /anvil/projects/tdm/data/icecream/combined/reviews.csv

5277


4754 lines include the exact pattern "chocolate".
5277 lines include the pattern "chocolate" as a case-insensitive search.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.